packages/ui/src/helpers/authenticator/utils.ts

```diff
@@ -153,3 +153,13 @@ export const getTotpCodeURL = (
   encodeURI(
     `otpauth://totp/${issuer}:${username}?secret=${secret}&issuer=${issuer}`
   );
+
+export const trimValues = (
+  formData: Record<string, string>,
+  ...ignoreList: string[]
+): Record<string, string> => {
+  return Object.entries(formData).reduce((trimmedFormData, [name, value]) => {
+    trimmedFormData[name] = ignoreList.includes(name) ? value : value?.trim();
+    return trimmedFormData;
+  }, {});
+};
```

Comment:
What do you think of adding generic support for the values to trim, e.g.
```suggestion
export function trimValues<T extends Record<string, string>>(
  values: T,
  ...ignored: string[]
): T {
  return Object.entries(values).reduce(
    (acc, [name, value]) => ({
      ...acc,
      [name]: ignored.includes(name) ? value : value?.trim(),
    }),
    {} as T
  );
}
```

This would allow us to keep the typing in place in scenarios like:

<img width="603" alt="image" src="https://user-images.githubusercontent.com/15002885/211935302-bdd39f95-ffd8-4dc1-b61d-e69f353ad69b.png">

Reply:
SGTM 😈